File: leet_code_tranining/programming_exercices/2_average_salary.js



var average = function(salary) {

    let minValue = Math.min(...salary);
    let maxValue = Math.max(...salary);
    console.log(`minValue ${minValue} maxValue: ${maxValue}`)
    let total = 0;
    for (let i = 0; i<salary.length; i++) {
        sal = salary[i]
        if (sal == minValue || sal ==maxValue){
            continue;
        }
        console.log(`adding ${sal}`)
        total += sal
        
    }
    result = total / (salary.length - 2)
    return result.toFixed(5)
};

if (require.main === module) {

  
    let inputs = [4000,3000,1000,2000];
    // 2500.00000
    console.log(average(inputs));
    // 2000.00000
    inputs = [1000,2000,3000];
    console.log(average(inputs));
    // 4750.00000
    inputs = [8000,9000,2000,3000,6000,1000];
    console.log(average(inputs));
  }

File: leet_code_tranining/programming_exercices/2_average_salary.py
from typing import List
import math
from decimal import Decimal, getcontext

class Solution:
    def average(self, salary: List[int]) -> float:
        sorted_salaries = sorted(salary)[1:-1]
        average_num = sum(sorted_salaries)/len(sorted_salaries)
        return  Decimal(str(average_num)).quantize(Decimal('0.00001'))


if __name__ == "__main__":
    s = Solution()

    inputs = [4000,3000,1000,2000]
    print(s.average(inputs))
    inputs = [1000,2000,3000]
    print(s.average(inputs))
    inputs = [8000,9000,2000,3000,6000,1000]
    print(s.average(inputs))
